src/traces/sankey/plot.js

```diff
@@ -0,0 +1,243 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var render = require('./render');
+var Fx = require('../../plots/cartesian/graph_interact');
+var d3 = require('d3');
+var Color = require('../../components/color');
+
+function renderableValuePresent(d) {return d !== '';}
+
+function ownTrace(selection, d) {
+    return selection.filter(function(s) {return s.key === d.traceId;});
+}
+
+function makeTranslucent(element, alpha) {
+    d3.select(element)
+        .select('path')
+        .style('fill-opacity', alpha);
+    d3.select(element)
+        .select('rect')
+        .style('fill-opacity', alpha);
+}
+
+function makeTextContrasty(element) {
+    d3.select(element)
+        .select('text.name')
+        .style('fill', 'black');
+}
+
+function relatedLinks(d) {
+    return function(l) {
+        return d.node.sourceLinks.indexOf(l.link) !== -1 || d.node.targetLinks.indexOf(l.link) !== -1;
+    };
+}
+
+function relatedNodes(l) {
+    return function(d) {
+        return d.node.sourceLinks.indexOf(l.link) !== -1 || d.node.targetLinks.indexOf(l.link) !== -1;
+    };
+}
+
+function nodeHoveredStyle(sankeyNode, d, sankey) {
+
```

Comment:
:hocho: those blank line at the start of new blocks in accordance with standard style.